src/components/layout/Undernames/Undernames.tsx

```diff
@@ -0,0 +1,188 @@
+import { Pagination, Tooltip } from 'antd';
+import Table from 'rc-table';
+import { ColumnType } from 'rc-table/lib/interface';
+import { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useArweaveCompositeProvider, useIsMobile } from '../../../hooks';
+import { useUndernames } from '../../../hooks/useUndernames/useUndernames';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  UndernameMetadata,
+} from '../../../types';
+import { isArweaveTransactionID } from '../../../utils';
+import eventEmitter from '../../../utils/events';
+import { ArrowLeft } from '../../icons';
+import Loader from '../Loader/Loader';
+
+function Undernames() {
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const isMobile = useIsMobile();
+  const navigate = useNavigate();
+  const { id } = useParams();
+  if (!id) {
+    eventEmitter.emit('Cannot load Undernames, no PDNT ID was found');
```

Comment:
not sure we should even emit this

Reply:
I figured if we are rerouting them its good to show an error about why.

Reply:
Fair. the format of this is wrong.

It should be 

`eventEmitter.emit('error', new Error('Missing PDNT transaction ID.')`